### PR TITLE
proposal for an enforcer config file

### PR DESCRIPTION
**Pull request by jbtrystram**

### CODE REVIEW

File: seedwing-policy-cli/README.md [hunk 1]
```diff
@@ -93,3 +94,18 @@ Rationale:
 ok!
 ```
 
+### Using a configuration file
+
+The following config file end up applying the same arguments showed above :
+```toml
+[enforcer]
+[enforcer.evaluate]
```
Comment (lulf) [inline seedwing-policy-cli/README.md:102]: Maybe replace 'evaluate' with 'eval'.

File: seedwing-policy-cli/README.md [hunk 2]
```diff
@@ -93,3 +94,18 @@ Rationale:
...
+policy="yaml-rule.dog"
+requires="yaml-rule::nr"
+
+[enforcer.verify]
+policy="path/to/policy"
+```
+
+Seedwing policy cli `evaluate` and `verify` commands will look for a `.enforcer.toml` file in the current working directory and apply it if no
```
Comment (lulf) [inline seedwing-policy-cli/README.md:110]: I wonder if we should make this into a '.seedwing.toml' file, and essentially contain contents like this:

```
[policies]
policy_dir = "my-policy-folder"
patterns = [
  "my::policy::this",
  "my::policy::that"
]
```

Comment (bobmcwhirter) [reply seedwing-policy-cli/README.md:110]: If so `policy_dir` should be plural-able. From the cli we can `-p` multiple times.

**Comment by jbtrystram** [comment]
@ctron I'm not really sure what I should do here. Do you think it makes sense to bring `config.rs` from the enforcer repo here ?

**Comment by ctron** [comment]
I am not a fan of using `toml`. I think that `.yaml` is much more common. And having an extension of `.yaml` helps editors to understand what kind of file this is. So I think it should be something like `.foobar.yaml`.

I am also not completely sure I understand what the goal is. Right now we have the "enforcer" and the "policy" repository. "policy" containing the "engine" and the "server". Not the enforcer relies on the "engine", so pulling in its code will also pull in all other kind of dependencies (like the LSP stuff) which I am not sure I would like to see here (maybe others agree).

Also I think that the "enforcer" is a tad different than the "swio". And I don't yet understand if that should be the case, or should change. And I think for this it needs a bit more experimenting.

For example:

My idea of the enforcer is to be as end-user IDE/CI friendly as possible. While I would expect the `swio` binary to be a much more generic thing.

If we continue with the enforcer, we would definitely need to put in some understanding of the code structure it resides in. Like where are top-level projects inside the repo, why type are they (auto, maven, node, cargo, …). Ideally stuff get's auto-detected, but if that leads to wrong results, there should be a way to override.

I am not sure if all of this information belongs into the context of `swio`. However, I also see a good reason to make `swio` the swiss-army-knife that can do everything.

Then again, it we start out in the enforcer repository, we should easily be able to, later on, pull in the code into this repo. And I guess this would allow the -policy repo more room to evolve, and give the -enforcer repo a bit more stability to work on top if it.

**Comment by jbtrystram** [comment]
Allright, let me try to sum up what we decided: 
- work to make the enforcer have a "run-once" mode
- Migrate the GH action to use the enforcer
- Add support for other things than maven `pom.xml` 

See later on if we want to bring that code in `swio` or not.

**Pull request closed without merging**